### PR TITLE
Add dql option to set custom DQL

### PR DESCRIPTION
**Pull request by Seb33300**

### CODE REVIEW

**Comment by Seb33300** [comment]
Fix https://github.com/stwe/DatatablesBundle/issues/525

**Comment by stwe** [comment]
Looks like a "Killer feature". Give me a few days to fully understand the code. Thank you in advance.

**Comment by Seb33300** [comment]
Just add comments on my code if you don't understand some parts

**Comment by mssimi** [comment]
What about using property accessor instead eval()?

**Comment by Seb33300** [comment]
@mssimi Thanks! I was looking for a better way to do this but didn't found.
Didn't know that PropertyAccess whorks with array.
I am also cleaning the old array key too now.

File: Datatable/Column/AbstractColumn.php [hunk 1]
```diff
@@ -752,13 +760,14 @@ public function getDql()
...
      *
      * @return $this
      * @throws Exception
      */
-    public function setDql($dql)
+    public function setDql($dql, $validate = false)
     {
-        if (true === $this->dqlConstraint($dql)) {
+        if ((false == $validate) || (true === $this->dqlConstraint($dql))) {
             $this->dql = $dql;
```
Comment (stwe) [inline Datatable/Column/AbstractColumn.php]: call setDql('foo', true)
possibly: If validate = true, the result of the "dqlConstraint" function is irrelevant

Comment (Seb33300) [inline Datatable/Column/AbstractColumn.php]: if `$validate == true`, we want to validate DQL so check if `dqlConstraint` is true
if `$validate == false`, it's a custom DQL code so we don't check if `dqlConstraint` is true

**Comment by stwe** [comment]
Ok I will try the PR at next weekend. But I will copy & paste the code since the PR has some conflicts (my fault).

**Comment by Seb33300** [comment]
I resolved conflicts

**Comment by stwe** [comment]
ok let's try :)

**Comment by stwe** [comment]
My computer is not exploded. 👍  Thank you. I'll continue on the weekend.

**Comment by Seb33300** [comment]
Ahah. Good 👍

**Pull request merged**